app/src/test/java/tech/relaycorp/test/WaitAssertions.kt

```diff
@@ -0,0 +1,26 @@
+package tech.relaycorp.test
+
+import kotlinx.coroutines.delay
+
+object WaitAssertions {
+
+    suspend fun waitForAssertEquals(expected: Any, actualCheck: suspend () -> Any) {
```

Comment:
How about something like this to detect type errors at compile time?

```suggestion
    suspend fun <T> waitForAssertEquals(expected: T, actualCheck: suspend () -> T) {
```

Reply:
I started with that, but it made type checking harder for the compiler, and meant we couldn't use method references to keep the code a bit simpler.

Reply:
:+1: makes sense